osm-fudge/bk_tree.py

```diff
@@ -0,0 +1,42 @@
+'''An implementation of bk-tree
+
+More: https://en.wikipedia.org/wiki/BK-tree
+'''
+
+class BKTree:
+    '''A bk-tree implementation and defined user methods
+       for insertion and querying the string
+    '''
+
+    def __init__(self, root, distanceMetrics, options):
+        self.tree = {root: {}}
+        self.distanceMetrics = distanceMetrics
+        self.options = options
```

Comment:
As stated above the metric options are not really a good fit for the bktree. They should be bound to the metric in the first place. The bktree only wants a distance metric :)

Reply:
dunzo 👍